scripts/check_openshift_version.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import json
+import logging
+from http.client import HTTPResponse
+from pathlib import Path
+from typing import Dict, List
+from urllib import request
+from urllib.error import HTTPError
+
+import semver
+
+ROOT = Path(__file__).resolve().parent.parent
+FILES_IN_PELORUS = [
+    ROOT / ".github/ISSUE_TEMPLATE/bug.yml",
+    ROOT / "docs/UpstreamSupport.md",
+]
+FILES_IN_OPENSHIFT = [
+    "ci-operator/jobs/dora-metrics/pelorus/dora-metrics-pelorus-master-presubmits.yaml",
+]
+OPENSHIFT_REPO = "openshift/release"
+OPENSHIFT_BRANCH = "master"
+RAW_URL = f"https://raw.githubusercontent.com/{OPENSHIFT_REPO}/{OPENSHIFT_BRANCH}"
+
+
+def get_supported_versions() -> List[str]:
+    with request.urlopen(
+        "https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/graph"
+    ) as response:
+        versions: List[Dict[str, str]] = list(json.load(response)["nodes"])
+    stable_versions = [
+        version["version"]
+        for version in versions
+        if not version["version"].upper().isupper()
+    ]
+    minor_versions = {
+        semver.VersionInfo.parse(version).replace(patch=0)
+        for version in stable_versions
+    }
+    latest_minor_versions = list(minor_versions)
+    latest_minor_versions.sort(reverse=True)
+
+    return [f"{version.major}.{version.minor}" for version in latest_minor_versions[:4]]
+
+
+def check_version_in_openshift_repo(version: str, file_name: str) -> None:
+    response: HTTPResponse
```

Comment:
Why to add here a type? Unnecessary imo.

Reply:
did this because the return type of urlopen is broader then just HTTPResponse. This way, we are telling exactly what is the return type of the function

Reply:
It really doesn't guard against it:

```
def return_type() -> float:
  return 1.0
   
def my_func():
  a: str
  a = return_type()
  print(type(a))

my_func()
<class 'float'>
````

Reply:
it's useless imo as either really check for the type or do not include such line at all, but won't be fighting over it :)